vm_insnhelper.c

```diff
@@ -1173,7 +1173,7 @@ vm_getivar(VALUE obj, ID id, const rb_iseq_t *iseq, IVC ic, const struct rb_call
         }
 
         val = ivar_list[index];
-        VM_ASSERT(rb_ractor_shareable_p(obj) ? rb_ractor_shareable_p(val) : true);
+        VM_ASSERT(BUILTIN_TYPE(obj) == T_OBJECT && rb_ractor_shareable_p(obj) ? rb_ractor_shareable_p(val) : true);
```

Comment:
Does this assert actually do anything? It seems to be the same as `cond ? cond : true` which is the same as `cond ? true : true` and so just `true`.

Reply:
It's subtle, but the first call to `rb_ractor_shareable_p` has `obj` as an arg, and the second one has `val` as an arg. So it's not `cond ? cond : true`, it's more like `if cond1 { assert cond2 }`

Reply:
Oh right, thanks for clarifying, that makes more sense. (`val` should be `value` and it'd be trivial to see, oh well)